File: clients/android/AwareChat-Android/app/src/main/java/com/example/awarechat_android/designsystem/components/LargeButton.kt
package com.example.awarechat_android.designsystem.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import com.example.awarechat_android.designsystem.tokens.ColorTokens
import com.example.awarechat_android.designsystem.tokens.SizeTokens
import com.example.awarechat_android.designsystem.tokens.SpacingTokens
import com.example.awarechat_android.ui.theme.AwareChatAndroidTheme

enum class LargeButtonStyle {
    PRIMARY,
    SECONDARY,
}

@Composable
fun LargeButton(
    text: String,
    style: LargeButtonStyle,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    val containerColor = when (style) {
        LargeButtonStyle.PRIMARY -> ColorTokens.primary
        LargeButtonStyle.SECONDARY -> ColorTokens.secondary
    }
    val contentColor = when (style) {
        LargeButtonStyle.PRIMARY -> ColorTokens.background
        LargeButtonStyle.SECONDARY -> ColorTokens.textPrimary
    }

    Button(
        onClick = onClick,
        modifier = modifier.height(SizeTokens.largeButtonHeight),
        shape = CircleShape,
        colors = ButtonDefaults.buttonColors(
            containerColor = containerColor,
            contentColor = contentColor,
        ),
        contentPadding = PaddingValues(horizontal = SpacingTokens.medium),
    ) {
        BodyText(
            text = text,
            fontWeight = FontWeight.Medium,
            color = contentColor,
        )
    }
}

@Preview(name = "Large Button", showBackground = true)
@Composable
private fun LargeButtonPreview() {
    AwareChatAndroidTheme {
        Column(
            modifier = Modifier.padding(SpacingTokens.medium),
            verticalArrangement = Arrangement.spacedBy(SpacingTokens.medium),
        ) {
            LargeButton(
                text = "Primary",
                style = LargeButtonStyle.PRIMARY,
                onClick = {},
                modifier = Modifier.fillMaxWidth(),
            )
            LargeButton(
                text = "Secondary",
                style = LargeButtonStyle.SECONDARY,
                onClick = {},
                modifier = Modifier.fillMaxWidth(),
            )
        }
    }
}

File: clients/ios/AwareChat-iOS/AwareChat-iOS/DesignSystem/Components/LargeButton.swift
//
//  LargeButton.swift
//  AwareChat-iOS
//

import SwiftUI

struct LargeButton: View {
  enum Style {
    case primary
    case secondary
  }

  let title: String
  let style: Style
  let action: () -> Void

  init(
    _ title: String,
    style: Style,
    action: @escaping () -> Void
  ) {
    self.title = title
    self.style = style
    self.action = action
  }

  var body: some View {
    Button(action: action) {
      BodyText(title, weight: .medium)
        .foregroundStyle(foregroundColor)
        .frame(maxWidth: .infinity, maxHeight: .infinity)
    }
    .frame(height: Tokens.Size.LargeButtonHeight.value)
    .background(backgroundColor)
    .clipShape(Capsule())
  }

  private var backgroundColor: Color {
    switch style {
    case .primary:
      Tokens.Colors.primary
    case .secondary:
      Tokens.Colors.secondary
    }
  }

  private var foregroundColor: Color {
    switch style {
    case .primary:
      Tokens.Colors.background
    case .secondary:
      Tokens.Colors.textPrimary
    }
  }
}

#Preview("Large Button") {
  VStack(spacing: Tokens.Spacing.medium.value) {
    LargeButton("Primary", style: .primary) { }
    LargeButton("Secondary", style: .secondary) { }
  }
  .padding(Tokens.Spacing.medium.value)
  .background(Tokens.Colors.background)
}
